hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -161,66 +162,70 @@ public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
                                    Option<InstantRange> instantRange,
                                    TaskContextSupplier taskContextSupplier,
                                    CompactionExecutionHelper executionHelper) throws IOException {
-    HoodieStorage storage = metaClient.getStorage();
-    Schema readerSchema;
-    Option<InternalSchema> internalSchemaOption = Option.empty();
-    if (!StringUtils.isNullOrEmpty(config.getInternalSchema())) {
-      readerSchema = new Schema.Parser().parse(config.getSchema());
-      internalSchemaOption = SerDeHelper.fromJson(config.getInternalSchema());
-      // its safe to modify config here, since we are running in task side.
-      ((HoodieTable) compactionHandler).getConfig().setDefault(config);
+    if (config.getBooleanOrDefault(HoodieReaderConfig.FILE_GROUP_READER_ENABLED)
+        && compactionHandler.supportsFileGroupReader()) {
```

Comment:
Since we are writing this in the last min, would like to keep the new compaction only for partial update flow. So, it could mean the following
if new FG reader is enabled and do one parsing of log headers to find if there are any partial update log blocks. If yes, we go w/ new way of compacting. if not, fallback to old way of compacting. 

We can do more testing and flip it in 1.0.1 may be. But feels risky to do it for 1.0

Reply:
>keep the new compaction only for partial update flow

by that, we mean if the compaction has at least 1 log file with partial updates? its hard to figure this out without reading the headers, right? .. All of the snapshot queries are going through the same code path (fg reader). So we should bite the bullet right now and stabilizing in 1.0.1, if there are bugs..

Reply:
Can we add a control option to by defaut enable the new fg reader compaction, so the user at least can has a choice to fallback to old compaction for Spark if any regressions are encountered.

Reply:
I agree with @vinothchandar that major use cases of snapshot queries already work with file group reader.  There are cases that might have issues, e.g., MDT, schema on read, etc., which I have explicitly turned off the new feature to let them go through the old flow of compaction.

@danny0405 yes, if there is any regression in Spark, user can turn off fg reader based compaction by turning off `HoodieReaderConfig.FILE_GROUP_READER_ENABLED` / `hoodie.file.group.reader.enabled`.